torch/csrc/jit/tracer.cpp

```diff
@@ -39,11 +39,57 @@ thread_local std::shared_ptr<TracingState> tracing_state;
 
 } // namespace detail
 
+// Given a variable 'var', return the 'node' which represents the instruction
+// which computes the value of this variable in the IR.
+// Here, we interpret untraced variables as constants that are just embedded
+// in the graph.  This is useful to handle code which does things like this
+// (from torch.autograd.variable, now moved to C++):
+//
+//    def mm(self, matrix):
+//      output = Variable(self.data.new(self.data.size(0), matrix.data.size(1)))
+//      return Addmm.apply(output, self, matrix, 0, 1, True)
+//
+// Here, mm fakes up a dummy variable with uninitialized data to do an inplace
+// update on, but subsequently ignores it because the alpha scaling factor is
+// zero. This is one of the cases where a Variable can be created inside of a
+// trace, and if we treat it as a constant, everything will work out.
+Value* getValueTrace(const Variable& var) {
+  auto& state = getTracingState();
+  if (!var.defined()) {
+    Node* n = state->graph->createUndefined();
+    return state->graph->insertNode(n)->output();
+  }
+
+  auto& env_stack = getTracingState()->env_stack;
```

Comment:
This is repeated both in getValueTrace and getFutureTrace. It makes more sense to write a single get/setValueTrace that takes an IValue and then handles the future/Tensor distinction internally.